snapshot/snapshotter.go

```diff
@@ -256,74 +261,84 @@ func makeTest(t *testing.T, name string, snapshotterFn func(root string) (Snapsh
 	}
 }
 
+func tFatal(t *testing.T, err error) {
+	t.Fatal(errors.WithStack(err))
```

Comment:
`errors.WithStack` is just going to set the stack at this current location, which is not exactly what we want, calling `t.Fatal` directly already gives the failure location. Also using `errors.WithStack` will not cause the error to be formatted with the stack. I think it is cleaner to just use `t.Fatalf("failure reason: %+v", err)` rather calling a helper.